geth/transactions/addrlock.go

```diff
@@ -27,14 +29,16 @@ func (l *AddrLocker) lock(address common.Address) *sync.Mutex {
 	return l.locks[address]
 }
 
-// LockAddr locks an account's mutex. This is used to prevent another tx getting the
+// LockNonceFor locks an account's mutex and returns a nonce. This is used to prevent another tx getting the
 // same nonce until the lock is released. The mutex prevents the (an identical nonce) from
 // being read again during the time that the first transaction is being signed.
-func (l *AddrLocker) LockAddr(address common.Address) {
+func (l *AddrLocker) LockNonceFor(address common.Address) uint64 {
```

Comment:
Maybe `LockAddrAndReturnLocalNonce()`? As it's a bit hard to figure out if this method is just a getter or has some side-effects.

Or maybe just go with `Lock()`? I have some trouble with an idea of locking nonce as we actually lock an address and additionally keep track of its local nonce.